.gitignore

```diff
@@ -78,3 +78,5 @@ compose-dev.yaml
 # Do not ignore specific .txt files for license check
 !license-header.txt
 !license-style.txt
+
+*.prof
```

Comment:
I really don't like how we ignore all kinds of files like this. Yesterday, I ran into an issue with some code picking up the wrong `.so` file because I didn't realize that we were git-ignoring all `.so` files. This is a bad habit. Files that should be ignored should be put in specific folders. Can we stick the `.prof` files in a profiles directory where we ignore the contents?